Implemented Nurse Homepage

smithhug opened the pull request:
Implemented simple nurse homepage with "tasks" button. This will currently redirect to the patients list.

Also fixed the user select radio button to now login with Admin, Caretaker or Nurse depending what is selected. Unsure how long since this last worked but it is fixed here.

As this task will take some time I want to commit changes when each part is complete. See the Trello entry for "implement nurse user" for more.

smithhug committed:
Implemented Nurse Homepage

Implemented simple nurse homepage with "tasks" button. This will currently redirect to the patients list.

Also fixed the user select radio button to now login with Admin, Caretaker or Nurse depending what is selected. Unsure how long since this last worked but it is fixed here

As this task will take some time I want to commit changes when each part is complete. See the Trello entry for "implement nurse user" for more.

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.java:
```diff
@@ -0,0 +1,50 @@
+    // settings button
+    settingsButton.setOnClickListener(
+        view -> {
+          final Intent medicalDiagnosticsActivityIntent =
```
Could add this route to the NavigationService

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.java:
```diff
@@ -0,0 +1,50 @@
+        });
+
+    // sign out button
+    signOutButton.setOnClickListener(
```
Could add this route to the NavigationService

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.java:
```diff
@@ -0,0 +1,50 @@
+
+  @SuppressLint("MissingInflatedId")
+  @Override
+  protected void onCreate(final Bundle savedInstanceState) {
```
The logic here looks fine, but this file should be Kotlin

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.java:
```diff
@@ -0,0 +1,50 @@
+    settingsButton = findViewById(R.id.settingsButton_nurse);
+    signOutButton = findViewById(R.id.sighOutButton_nurse);
+
+    // patient list button
```
IMO the comment here and just below isn't that necessary, but I'll leave that up to you

el-veee reviewed (approved):
Looks pretty good. I've added some comments for minor improvements but ill approve pending those changes

smithhug committed:
Converted Homepage4nurse to kotlin

added route to settings and signout to navigationservice.

Changed Homepage4nurse to kotlin.

Decided to keep comments for my own personal reminders as I will continue to work on and develop the nurse user throughout the project. Will remove unnecessary comments once nurse is feature complete.

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.kt:
```diff
@@ -0,0 +1,39 @@
+import deakin.gopher.guardian.services.NavigationService
+
+class Homepage4nurse : AppCompatActivity() {
+    @SuppressLint("MissingInflatedId")
```
I'd say remove this and see if it still builds and runs. It might not be necessary

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.kt:
```diff
@@ -0,0 +1,39 @@
+        val signOutButton: Button = findViewById(R.id.sighOutButton_nurse)
+
+        // patient list button
+        tasksButton.setOnClickListener(
```
For this listener, you can use Kotlin's Lambda syntax where you remove the () and just use {} like the listeners below

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.kt:
```diff
@@ -0,0 +1,39 @@
+
+        // patient list button
+        tasksButton.setOnClickListener(
+            View.OnClickListener { view: View? ->
```
I don't think you'd need the View.OnClickListener block or the view: View? -> part. The syntax should work if you match the other listeners below

el-veee commented on app/src/main/java/deakin/gopher/guardian/view/general/Homepage4nurse.kt:
```diff
@@ -0,0 +1,39 @@
+        tasksButton.setOnClickListener(
+            View.OnClickListener { view: View? ->
+                val medicalDiagnosticsActivityIntent =
+                    Intent(this@Homepage4nurse, PatientListActivity::class.java)
```
Could also be a route in the Navigation Service

smithhug committed:
Made changes

Commented out any redirections to TasksList as it is still being implemented.

Removed the supress line as well.

smithhug committed:
Merge branch 'master' into master

smithhug committed:
Update NavigationService.kt

Removed reference to incomplete activity

smithhug committed:
Update LoginActivity.kt

Updated naming convention as it caused build error on github
